README.md

```diff
@@ -1,602 +1,387 @@
+
+> **Note: This is not a checklist.**
+> 
+> Ask a hundred developers how to test an application,
+> and you’ll get a hundred different answers.
+> Some may insist on exhaustively testing every single method on every model;
+> others may contend that since integration tests capture model behavior anyway,
+> much unit testing is redundant and burdensome.
+>
+> RSpec Rails provides thoughtfully selected features
+> to encourage good testing practices, but there’s no “right” way to do it.
+> Ultimately, it’s up to you to decide how your test suite will be composed.
+> 
+> (You may even define custom types, or write specs with no type at all.)
+
+[request]: https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec
+[feature]: https://www.relishapp.com/rspec/rspec-rails/docs/feature-specs/feature-spec
+[system]: https://relishapp.com/rspec/rspec-rails/docs/system-specs/system-spec
+[model]: https://www.relishapp.com/rspec/rspec-rails/docs/model-specs
+[controller]: https://www.relishapp.com/rspec/rspec-rails/docs/controller-specs
+[mailer]: https://relishapp.com/rspec/rspec-rails/docs/mailer-specs
+[job]: https://relishapp.com/rspec/rspec-rails/docs/job-specs/job-spec
+[view]: https://www.relishapp.com/rspec/rspec-rails/docs/view-specs/view-spec
+[routing]: https://www.relishapp.com/rspec/rspec-rails/docs/routing-specs
+[helper]: https://www.relishapp.com/rspec/rspec-rails/docs/helper-specs/helper-spec
+[`ActionDispatch::IntegrationTest`]: https://edgeapi.rubyonrails.org/classes/ActionDispatch/IntegrationTest.html
+[`ActionDispatch::SystemTestCase`]: https://edgeapi.rubyonrails.org/classes/ActionDispatch/SystemTestCase.html
+[`ActionController::TestCase`]: https://edgeapi.rubyonrails.org/classes/ActionController/TestCase.html
+
+### I’m writing a spec file from scratch. How do I assign it a type?
+
+Assuming you haven’t [modified the default `rails_helper.rb` configuration][],
+simply place the spec in the appropriate folder
+(_e.g.,_ `spec/models/` for model specs)
+and RSpec will set its type automatically.
+
+If you _have_ modified the default config
+(or if you just want to be extra explicit),
+you can set the `:type` option at the top of the file, like so:
 
 ```ruby
-# rspec-rails-1.x
-assigns[key] = value
+# spec/models/user_spec.rb
 
-# rspec-rails-2.x+
-assign(key, value)
+RSpec.describe User, type: :model do
+...
 ```
 
-### `rendered`
+[modified the default `rails_helper.rb` configuration]: https://relishapp.com/rspec/rspec-rails/docs/upgrade#file-type-inference-disabled
 
-This represents the rendered view.
+### System specs, feature specs, request specs–what’s the difference?
 
-```ruby
-render
-expect(rendered).to match /Some text expected to appear on the page/
-```
-
-#### Upgrade note
+RSpec Rails provides three types of specs
+for integration testing the application as a whole—in other words,
+specifying what the client sees when interacting with it.
```

Comment:
Not technically true, only feature / system specs have any javascript capability so they're the only ones capable of being end to end. Also integration tests don't have to be end to end, see earlier notes about confusing terminology.

Reply:
Understood.

How do you feel about the following distinction instead?

> RSpec Rails provides three types of specs that do not directly correspond to any Rails application component.

I think this was the source of my original confusion about what certain spec types were for, and is (for me) the main criterion that distinguishes which spec types are worth explaining in detail and which should simply be linked to the Cucumber docs.

Reply:
How about
```
RSpec Rails also provides some end to end (entire application) testing capability, to specify the interaction with the client.
```

Reply:
My understanding was that you didn't want request specs grouped under the end-to-end umbrella. Would you like me to extract the **Request Specs** subsection into its own, separate section, or are you okay with leaving them there?